src/rust/engine/fs/src/testutil.rs

```diff
@@ -0,0 +1,24 @@
+// Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+// Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+use std::fs::File;
+use std::io::Write;
+use std::path::Path;
+
+// NOTE: This method is not in the `src/rust/engine/testutil` crate because that will cause a cyclic
+// dependency between `fs` and `testutil` due to `testutil` using types from this `fs` crate in its
+// public interface.
+
+pub(crate) fn make_file(path: &Path, contents: &[u8], mode: u32) {
+    let mut file = File::create(path).unwrap();
+    file.write_all(contents).unwrap();
+
+    #[cfg(not(target_os = "windows"))]
```

Comment:
I notice that we don't have this attribute on the same function in testutil. Worth keeping them in sync?

Reply:
Probably. I took the liberty to improve this function as I copied it since I prefer platform-specific code to always be guarded (even if Pants does not currently work on Windows).

Reply:
Oh for sure, this is better. I meant is it worth adding this to the other function.

Reply:
I'll make the change this evening.

Reply:
Change made.